actorguidelines.md

```diff
@@ -0,0 +1,84 @@
+
+**script.ext**: Script file used by actor. A common practice is naming such script based on actor name. Another one is writing such script using Python or Bash script.
+
+It is a responsibility of actor’s writer to make sure that the actor’s directory contains all the necessary dependencies alongside the code. In other words, the actor should not rely on anything on the system on which the actor will be executed, except the following:
+Python 2.6+ installed
+
+README files or whatever is necessary to properly document an actor should be placed inside actor’s directory.
+
+
+## Naming Convention
+
+While creating a new actor or defining a new schema, is up to developer to name them in a way that will be easy for others to understand the functionality that such instance provides. From the name of each actor or schema it should be obvious not only which topic it handles, but also which specific part of the process. Examples are:
+
+**Actors**: i.e. “dns_server_config_check”, “web_server_config_fix” and “ldap_users_backup”.
+
+**Schemas**: Currently divided into two groups: single value of a basic type, named after that type (i.e  “number”, “string”, “string_list” or “bool”) and composed ones, named after the representative data (i.e “redis”, “osversion” or “process_list”).
+
+Currently there is no hard rules about naming but some soft conventions are used:
+* Use descriptive names;
+* Use snake_case wherever it is necessary;
+* Do not start a schema name with “type_” since that is implicit;
+* Do not use “actor” in an actor name since that is implicit.
+
+
+## Coding Guidelines
+
+Currently all existing actors are written using either Python or Bash script. When using Python you can also follow general Python guidelines for Leapp projects:
+https://github.com/leapp-to/guidelines/blob/master/python.md
+
+Also make sure that new code does not have any lint errors when possible, i.e run pep8, pylint, flake8 on your Python code. Automate testing of your code is always a good policy but how to properly implement that for actors is an open question.
+
+
+## Actors Categorization by Tags
+
+Actors will be categorized according to the phase of the upgrade process each actor belong to.
+This may be a subject to change depending on Channel Data Model implementation details.
+
+For the moment the phases are defined as :
+* **Pre**
+* **Upgrade**
+* **Post**
+
+Additional tags: ( tentative)
+* **Info** ( or **Related**) - Denotes the actors, that do not affect upgrade process directly, but provide additional useful information. There may be a CLI/GUI option to filter this info out from the report.
+
+This categorization will allow running the scan phase separately by choosing CLI/GUI option.
+Each actor will report a message to log. Part of the message will be the actor’s tag.
+
+
+## Actors Categorization by Result or Severity
```

Comment:
This section covers just results, not severity - remove it from the title.

Reply:
Done.